tests/components/starlink/test_diagnostics.py

```diff
@@ -0,0 +1,32 @@
+"""Tests for Starlink diagnostics."""
+import json
+
+from homeassistant.components.starlink.const import DOMAIN
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+
+from .patchers import COORDINATOR_SUCCESS_PATCHER
+
+from tests.common import MockConfigEntry, load_fixture
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+from tests.typing import ClientSessionGenerator
+
+
+async def test_diagnostics(
+    hass: HomeAssistant, hass_client: ClientSessionGenerator
+) -> None:
+    """Test generating diagnostics for a config entry."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_IP_ADDRESS: "1.2.3.4:0000"},
+    )
+
+    with COORDINATOR_SUCCESS_PATCHER:
+        entry.add_to_hass(hass)
+
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        diag = await get_diagnostics_for_config_entry(hass, hass_client, entry)
+
+        assert diag == json.loads(load_fixture("diagnostics_expected.json", "starlink"))
```

Comment:
You could use snapshots here instead, see: <https://developers.home-assistant.io/blog/2023/02/20/snapshot-testing>

Reply:
Done, thanks!